src/azure-cli/azure/cli/command_modules/role/tests/latest/test_role.py

```diff
@@ -679,6 +634,54 @@ def check_changelogs():
                 self.cmd('ad user delete --id {upn}')
 
 
+class RoleAssignmentWithConfigScenarioTest(RoleScenarioTestBase):
+
+    def __init__(self, *arg, **kwargs):
+        super().__init__(*arg, random_config_dir=True, **kwargs)
+
+    @ResourceGroupPreparer(name_prefix='cli_role_assign')
+    @AllowLargeResponse()
+    def test_role_assignment_handle_conflicted_assignments(self, resource_group):
```

Comment:
`test_role_assignment_handle_conflicted_assignments` is the only test that writes to `config` file. I moved it to a dedicated `RoleAssignmentWithConfigScenarioTest` so that other tests in `RoleAssignmentScenarioTest` won't need `random_config_dir`.